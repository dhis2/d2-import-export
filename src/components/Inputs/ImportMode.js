import i18n from '@dhis2/d2-i18n'
import React from 'react'
import { RadioGroup } from '../FinalFormComponents/RadioGroup'

export const OPTION_YES = { value: 'VALIDATE', label: i18n.t('Yes') }
export const OPTION_NO = { value: 'COMMIT', label: i18n.t('No') }
export const IMPORT_MODE_KEY = 'importMode'
export const IMPORT_MODE_DEFAULT_VALUE = OPTION_NO.value

/**
 * This input is used only for meta data imports
 * It's supposed to look like the dry run option
 */
export const ImportMode = () => (
    <RadioGroup
        name={IMPORT_MODE_KEY}
        label={i18n.t('Dry run')}
        options={[OPTION_YES, OPTION_NO]}
    />
)
