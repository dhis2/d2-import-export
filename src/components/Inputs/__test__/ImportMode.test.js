import React from 'react'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import { ImportMode } from '../ImportMode'

jest.mock('react-final-form')

describe('Input component - ImportMode', () => {
    it('should render correctly', () => {
        const file = shallow(<ImportMode />)

        expect(toJson(file)).toMatchSnapshot()
    })
})
