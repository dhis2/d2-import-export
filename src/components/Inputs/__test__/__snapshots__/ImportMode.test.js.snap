// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component - ImportMode should render correctly 1`] = `
<Field>
  <Label>
    Dry run
  </Label>
  <RadioGroup
    dataTest="input-import-mode"
    label="Dry run"
    name="importMode"
    options={
      Array [
        Object {
          "label": "Yes",
          "value": "VALIDATE",
        },
        Object {
          "label": "No",
          "value": "COMMIT",
        },
      ]
    }
  />
</Field>
`;
